### DISCUSSION TITLE
v6.11 kernel testing

### DISCUSSION THREAD

**Discussion by hexdump0815** [velvet-os/imagebuilder#243] [General]
most probably next weekend (but maybe already this one) the v6.11 kernel is supposed to be released and it is most probably the last version before the next lts kernel, which should be v6.12 this year and which i would like to use as base for most of the next round of images as it is lts, so it will have support for quite a while. because of that it might be a good time to start testing v6.11 on the different systems to see if there are any regressions, so that the chance of having a useable and well working v6.12 lts kernel end of the year is good.

i plan to create some documentation with suggestions of how such kernel testing could be done, what to test, what to keep in mind while doing it and what to do if things do not work as expected (i.e. patches not applying properly anymore or the resulting kernel not booting or working at all etc.). the question now would be: who would be interested to join the party and for which system(s)? the more coverage of the different systems we can achieve the better. once we start with it (i would suggest to maybe start when v6.11.1 is out) we can then use this discussion here to exchange results and help each other solving potential problems.

best wishes - hexdump

**Comment by thenameisluk** [comment]
i'm able to join on
trogdor
- homestar (duet 5)
- wormdingler (duet 3)
- lazor (spin 513)

kukui
- kodama (10e)
- krane (duet)

corsola
- magneton (slim 3)

**Comment by thenameisluk** [comment]
linux 6.11 is here :3
```
root@chluk:/compile/source/linux-stable-cbq# git pull
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 4 (delta 2), reused 0 (delta 0), pack-reused 0
Rozpakowywanie obiektów: 100% (4/4), 829 bajtów | 69.00 KiB/s, gotowe.
Z https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable
   d42f7708e27c..98f7e32f20d2  master     -> origin/master
 * [nowy tag]                  v6.11      -> v6.11
```

**Comment by thenameisluk** [comment]
# Lenovo Ideapad duet 5 (Homestar) 6.11.0-stb-cbq+


- wifi : works
- builtin display : works
- energy: battery status : works
- usb c for basic usb functionality : works
- gpu : works both opengl and vulkan
<img src="https://github.com/user-attachments/assets/59026605-20ba-4dee-b188-dd5fde89aab0" width="300">

- bluetooth : works (tested on headphones and mouse)
- energy: dpms, suspend/resume : works

~seams to break brightness settings (that was gnome 47 issue)~ there are some errors dmesg:
```
[ 4007.182013] [dpu error]connector not connected 3
[ 4007.182044] [drm:drm_mode_config_helper_resume] *ERROR* Failed to resume (-22)
```
- external display : works
<img src="https://github.com/user-attachments/assets/8e58fdb0-5f82-4ad3-bd99-3c55faf1b3d8" width="300">

- hw video encode/decode : decoding works
- audio : still [partially broken](https://github.com/hexdump0815/imagebuilder/discussions/240)

_Note. tested on debian trixie_

# additional notes
### broken usb?
i am aware of [usb disconnect issue](https://github.com/hexdump0815/imagebuilder/issues/47#issuecomment-2237290504), it is/seams to be present when basing ```.config``` of of ```config-6.6.23-stb-cbq``` or ```config-6.9.9-stb-cbq+``` and likely other too
_Note. tested lately on config-6.9.9-stb-cbq+ -> Image-6.11.0-rc7-stb-cbq+_

but creating config the way described in the [notes](https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel/blob/master/readme.cbq) solves the issue
### additionall config
i used additional configs when generating .config
/compile/bonus/waydroid.cfg for working waydroid
```
CONFIG_PSI=y
CONFIG_CHECKPOINT_RESTORE=y
CONFIG_RELR=y
CONFIG_PACKET_DIAG=m
CONFIG_UNIX_DIAG=y

CONFIG_NFT_NUMGEN=m
CONFIG_NFT_CT=m
CONFIG_NFT_FLOW_OFFLOAD=m
CONFIG_NFT_CONNLIMIT=m
CONFIG_NFT_LOG=m
CONFIG_NFT_LIMIT=m
CONFIG_NFT_MASQ=m
CONFIG_NFT_REDIR=m
CONFIG_NFT_NAT=m
CONFIG_NFT_TUNNEL=m
CONFIG_NFT_QUEUE=m
CONFIG_NFT_QUOTA=m
CONFIG_NFT_REJECT=m
CONFIG_NFT_REJECT_INET=m
CONFIG_NFT_COMPAT=m
CONFIG_NFT_HASH=m
CONFIG_NFT_XFRM=m
CONFIG_NFT_SOCKET=m
CONFIG_NFT_OSF=m
CONFIG_NFT_TPROXY=m
CONFIG_NFT_SYNPROXY=m
CONFIG_NFT_DUP_NETDEV=m
CONFIG_NFT_FWD_NETDEV=m
CONFIG_NFT_REJECT_NETDEV=m
CONFIG_NFT_REJECT_IPV4=m
CONFIG_NFT_REJECT_IPV6=m
CONFIG_NETLINK_DIAG=m
CONFIG_PROC_CHILDREN=y
```
/compile/bonus/gpm.cfg for working mouse inside a tty
```
CONFIG_INPUT_MOUSEDEV=y
```
but lack of these shouldn't break anything
### some warnings
```
make olddefconfig
```
throws some warnings
```
.config:1676:warning: symbol value 'n' invalid for BLK_DEV_RAM_SIZE
.config:2537:warning: override: reassigning to symbol TYPEC_QCOM_PMIC
.config:2543:warning: override: reassigning to symbol DRM_MSM
.config:2544:warning: override: reassigning to symbol BACKLIGHT_PWM
#
# configuration written to .config
#
```
but these don't seam to cause any issues
### build distribution
if someone is interested in testing this version
i will be putting prebuilt version for trogdor devices (this image and likely be use on other trogdor devices as well) inside [velvet repo](https://github.com/hexdump0815/imagebuilder/issues/209#issuecomment-2332433231) soon to make testing it easier on me (and others interested)

i will also likely be sharing kukui (and corsola if possible) build there for ppl who are just interested in testing these without compiling
_Important. these will require you to install [velvet tools](https://github.com/thenameisluk/velvettools), which by default flashes images parmamently after succesful boot but here it would be recommended to set flash option to manual in case usb issue comes back_

**Comment by hexdump0815** [comment]
@thenameisluk  - thanks a lot for that quick testing ... i'll look at the details the next days (no time right now) ... i'm currently trying to track down a problem which seems to break the display on krane for everything higher than v6.6 (other kukui devices with a different display setup seem to be better) - i'm already in the process of bisecting it and hope to know more until next weekend

**Reply by thenameisluk** [reply]
@hexdump0815 i have issue with some patches not applying and kernel not compiling for kukui
was trying to test kodama but 'm kinda stuck
since you are working on karen i assume you've dealt with this problem

do you have a new patches i should apply instead or know which patches i should use for now?

will experiment and try with just using patches that don't fail for now

**Comment by thenameisluk** [comment]
# Testing on other trogdor and kukui device
if someone wants to test 6.11 kernel on other **trogdor** and **kukui** devices (builds for other chromebooks aren't there yet)
and share feedback
without building from source

1. add [velvet repo](https://github.com/thenameisluk/velvet-repo)
```
#add public key
curl -sS https://repo.velvet-os.org/repo/velvet_repo.asc | sudo tee -a /etc/apt/trusted.gpg.d/velvet_repo.asc
#add source
echo "deb [arch=arm64,all] https://repo.velvet-os.org/repo stable main" | sudo tee /etc/apt/sources.list.d/velvet_repo.list
```
2. update respositories
```
sudo apt update
```
3. install 6.11 kernel
 - on trogdor
```
sudo apt install linux-6.11.0-stb-cbq+ 
```
 - on kukui
```
sudo apt install linux-6.11.0-stb-mt8+
```
_Note. this will install [velvet tools](https://github.com/thenameisluk/velvet-repo) as dependency_
_Note. the build provided here is the same one [tested homestar](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10653865)_

4. open config file
```
sudo nano /etc/velvettools/config 
```
and change ```flash=auto``` to ```flash=manula``` to prevent velvet tools from autoflashing after succesfull boot

5. build bootable image
- on trogdor
```
sudo vtbuild 6.11.0-stb-cbq+ 
```
- on kukui
```
sudo vtbuild 6.11.0-stb-mt8+
```
_Note. if the command throws out of memory error you need to get rid of unused kernel's inside ```/boot```_

6. flash kernel on test partition
- on trogdor
```
sudo vttest 6.11.0-stb-cbq+
```
- on kukui
```
sudo vttest 6.11.0-stb-mt8+
```
confirm the partition is correct (on trogdor test partition should be ```/dev/mmcblk1p2```)

_Note. this will make it so for one boot your device will boot 6.11 kernel_
_Tip. if you want to reboot into that kernel again you need to run ^ command again or ```sudo vtflash ...``` if you want to always boot this version_
<details><summary>
Note. if the image is too big to flash
</summary>

it is very unlikely but just in case
open /etc/initramfs-tools/conf.d/compress
```
nano /etc/initramfs-tools/conf.d/compress
```
and put
```
COMPRESS=xz
XZ_OPT='-9 --check=crc32 –memlimit-compress=25%'
```
inside

then rebuild the image again with vtbuild

and try to flash it again
</details>

7. reboot your system
```
sudo reboot
```
<details><summary>
Note. if your system boots into weird terminal
</summary>

velvet tools when generating cmdline assume that your root partition label is ```rootemmc``` ~(this is a bug and will be fixed)~ (was fixed in version 0.6.8
to fix it, reboot your system with power button (unless you used vtflash your system should be back to normal)

remove ```/boot/cmdline``` and go back to step 5.
</details>

8. and check the kernel version

```
uname -a
```
the output should be similar to 
```
Linux cluk 6.11.0-stb-cbq+ #13 SMP PREEMPT Sun Sep 15 20:29:34 CEST 2024 aarch64 GNU/Linux
```
test how the kernel works and share what you think (feedback on velvet tools also apriciated)

good lu(c)k :3

**Reply by gallee** [reply]
FYI I tried `vttools` on my Juniper Kukui : however I did not get a boot (I guess because my rootemmc is luks encrypted).
Is it supported ?

**Reply by thenameisluk** [reply]
@gallee 
it should work and is supported

if you help me diagnose the issue i can fix it

could you share your blkid output
and /boot/cmdline

also did you have already working luks setup before? (without velvet tools)

**Reply by gallee** [reply]
Hello,
I am currently running 6.6.9-stb-mt8+ with luks on / 
Below the requested info 

* /boot/cmdline
`console=tty1 root=LABEL=rootemmc rootwait ro fsck.fix=yes fsck.repair=yes net.ifnames=0 ipv6.disable=1 quiet splash`

* blkid
```bash
/dev/mapper/encrypted: LABEL="rootemmc" UUID="53c70655-ca27-4bfa-80c4-734ae072e9b1" UUID_SUB="88de6922-3a5f-4cf7-aa05-dbe23b272209" BLOCK_SIZE="4096" TYPE="btrfs"
/dev/mmcblk0p3: LABEL="bootemmc" UUID="b077fad4-d2d6-4ccc-ab79-63056d33f7bc" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="09d12eda-6d0e-6b4b-b227-04190adaddd6"
/dev/mmcblk0p4: UUID="5e8ce006-4171-4deb-9079-f78dcd273328" TYPE="crypto_LUKS" PARTUUID="ee251dd6-22f5-6e4a-aba2-6e41460861b2"
```

**Reply by thenameisluk** [reply]
@gallee cmdline seams to ok
could you share ```/etc/fstab```, ```/etc/crypttab``` and ```/boot/dtbs``` (if exists)
and ```vtbuild``` and ```vttest``` output
also what does your device boot to? black screen?

**Reply by gallee** [reply]
@thenameisluk sorry for the delay ! I tried to test 6.13 but got same issue

/etc/fstab
```
LABEL=bootemmc /boot ext4 defaults,noatime,nodiratime,errors=remount-ro 0 2
LABEL=rootemmc / btrfs defaults,ssd,compress-force=zstd,noatime,nodiratime 0 1
/swap/file.0 none swap sw 0 0
```
/etc/crypttab
```
encrypted UUID=5e8ce006-4171-4deb-9079-f78dcd273328 none luks,discard
```
No /boot/dtbs
[vtbuilt output](https://gist.github.com/gallee/aa7df0b074a1bf6108ee66c9e62e352a)

vttest
```bash
sudo vttest 6.12.3-stb-mt8+
kernel chosen 6.12.3-stb-mt8+
flashing test partition
Are you sure you want to flash image to /dev/mmcblk0p2 ?
[y/N]
y
62808+0 records in
62808+0 records out
32157696 bytes (32 MB, 31 MiB) copied, 1.566 s, 20.5 MB/s

```

Boot gives me debian image and then 

![image](https://github.com/user-attachments/assets/a377b3df-a8de-4aee-abc6-2a4ec9a7bd3d)

**Reply by thenameisluk** [reply]
looks like the luks partition isn't getting mapped

very likely that for whatever reason initramfs isn't being included

can you make sure you have latest velvet tools installed

and try
```
update-initramfs -c -k 6.12.3-stb-mt8+
vtbuild 6.12.3-stb-mt8+ #can skip if it gets triggered automatically by initramfs update
vttest 6.12.3-stb-mt8+
```
share the full output
and checking if it works

**Reply by gallee** [reply]
I confirm with `update-initramfs` and vtest I am able to boot :rocket:

**Reply by thenameisluk** [reply]
nice
if everything boots correctly you can flash permanently with vtflash

**Reply by thenameisluk** [reply]
this one should no longer happen in version 0.7.0
https://github.com/thenameisluk/velvettools/commit/eb4c371a727cb12600b6edbd6ae660364a3d0c54

**Comment by thenameisluk** [comment]
# Acer chromebook spin sp513 (Lazor) 6.11.0-stb-cbq+

![IMG_5974](https://github.com/user-attachments/assets/79a794b9-2ac3-45c4-86b2-f3feb1d1193c)
- wifi : works
- builtin display : works
- energy: battery status : works
- usb c for basic usb functionality : works
- gpu : works both opengl and vulkan
- bluetooth : works (tested on headphones)
- energy: dpms, suspend/resume : works
there are some errors dmesg:
```
[ 4007.182013] [dpu error]connector not connected 3
[ 4007.182044] [drm:drm_mode_config_helper_resume] *ERROR* Failed to resume (-22)
```
- external display : works
- hw video encode/decode : decoding works
- audio : works

_Note. tested on debian trixie_

# additional notes

used velvet repo to test so [all homestar notes](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10653865) apply here

**Comment by thenameisluk** [comment]
# Lenovo ideapad duet 3 (Wormdingler) 6.11.0-stb-cbq+

![IMG_5975](https://github.com/user-attachments/assets/e894649b-9d52-404a-89d4-0ca2f7d1207b)
- wifi : works
- builtin display : works
- energy: battery status : works
- usb c for basic usb functionality : works
- gpu : works both opengl and vulkan
- bluetooth : works (tested on headphones)
- energy: dpms, suspend/resume : works
there are some errors dmesg:
```
[ 4007.182013] [dpu error]connector not connected 3
[ 4007.182044] [drm:drm_mode_config_helper_resume] *ERROR* Failed to resume (-22)
```
- external display : works
- hw video encode/decode : decoding works
- audio : works

_Note. tested on debian trixie_

# additional notes

used velvet repo to test so [all homestar notes](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10653865) apply here

**Comment by hexdump0815** [comment]
some updates from my side:
- my kukui krane display problem with v6.7 and newer bisecting ended nowhere and meanwhile i also know why :) ... i git bisected between v6.6 (good) and v6.7 (bad) and only bisected the drivers/gpu/drm path to reduce the bisection iterations by a few as i expected the problem somewhere there but that way i ended up at some commit in the nouveau/nvk driver which is not even compiled/used on kukui ... while looking closer at my display problem i noticed that xrandr display scaling (see: https://github.com/hexdump0815/imagebuilder/blob/main/files/extra-files/etc/lightdm/display-scaling.sh ) which i use on my krane seems to be somehow broken on newer kernels and results in a dark screen with backlight on, without xrandr display scaling everything is working fine on my krane and that is most probably also the reason why i did not see the problem on other kukui devices where i do not use xrandr display scaling ... the funny thing is that somehow i right now have my krane running with v6.11.0 and xrandr scaled display - no real idea right now how i got there while playing around with that ... in the end it seems there is no big problem with the display on kukui krane with newer kernels, which is very good of course - that xrandr display scaling thing we might look at later and i would not see this as a show stopper right now
- i have pushed my kukui v6.11 kernel patches and adjusted readme files to https://github.com/hexdump0815/linux-mainline-mediatek-mt81xx-kernel - that might be a good start for further testing ... for me it at least seems to boot and basic functionality seems to be there
- i have created a lts-v6.6 branch on the most used and mainline based kernel trees to have that state available somehow in case we for whatever reason want to build some more v6.6 lts kernels ... main/master is now where v6.11 and later v6.12 will happen now

**Comment by thenameisluk** [comment]
# Lenovo 10e (Kodama) 6.11.0-stb-mt8+

![IMG_5977](https://github.com/user-attachments/assets/50b52683-9c38-41dc-a8a8-e393dabd0269)

- wifi : works
- builtin display : works
- energy: battery status : works
- usb c for basic usb functionality : works
- gpu : works opengl
- bluetooth : works (tested on headphones)
- energy: dpms, suspend/resume : works
- external display : still broken
- hw video encode/decode : only via gpu ([mtk-vpu](https://wiki.postmarketos.org/wiki/Hardware_video_acceleration) doesn't seam to work)
```
mpv --hwdec=v4l2m2m-copy --hwdec-codecs=h264 Wideo/Kooha/Kooha-2024-05-16-15-29-34.mp4 
 (+) Video --vid=1 (*) (h264 1920x1200 60.000fps)
 (+) Audio --aid=1 (*) (mp3 2ch 48000Hz)
[ffmpeg/video] h264_v4l2m2m: Could not find a valid device
[ffmpeg/video] h264_v4l2m2m: can't configure decoder
Could not open codec.
```
- audio : works

_Note. tested on debian trixie_

# additional notes

### patches
build using [new patch set](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10663582)

### additional config
again used [additional homestar config files](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10653865) when generating .config

this shouldn't make any difference for the most part but should be note worthy

### build distribution
soon will update [repo](https://github.com/thenameisluk/velvet-repo) and [post](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10659997) to include kukui build

**Comment by thenameisluk** [comment]
# Lenovo ideapad duet (Krane) 6.11.0-stb-mt8+

![IMG_5978](https://github.com/user-attachments/assets/07cbdd95-cc33-4c72-aa4d-5a15793b3962)
- wifi : works
- builtin display : works
- energy: battery status : works
- usb c for basic usb functionality : works
- gpu : works opengl
- bluetooth : works (tested on headphones)
- energy: dpms, suspend/resume : works
- external display : still broken
- hw video encode/decode : only via gpu ([mtk-vpu](https://wiki.postmarketos.org/wiki/Hardware_video_acceleration) doesn't seam to work)
```
mpv --hwdec=v4l2m2m-copy --hwdec-codecs=h264 Wideo/Kooha/Kooha-2024-05-16-15-29-34.mp4 
 (+) Video --vid=1 (*) (h264 1920x1200 60.000fps)
 (+) Audio --aid=1 (*) (mp3 2ch 48000Hz)
[ffmpeg/video] h264_v4l2m2m: Could not find a valid device
[ffmpeg/video] h264_v4l2m2m: can't configure decoder
Could not open codec.
```
- audio : works

_Note. tested on debian trixie_

# additional notes

used [velvet repo](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10659997) to test so [all Kodama notes](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10663800) apply here

**Comment by thenameisluk** [comment]
# Testing on other devices
for anyone insterested in testing v6.11 kernel on kukui (or trogdor) chromebooks
i've update [the instructions](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10659997) and [repo](https://github.com/thenameisluk/velvet-repo?tab=readme-ov-file) to also include kukui kernel

for now i think these are all systems i could test it on (i will try to get corsola work on 6.11 but i'm not sure if it's doable for me), if anyone has device not tested yet or wants to test on their own device, go ahead!!!

**Comment by thenameisluk** [comment]
# Lenovo ideapad slim 3 (magneton) 6.11.0-stb-mt9

not booting/no screen output

# additional notes

based on stable kernel without additional patches

# Lenovo ideapad slim 3 (magneton) 6.10.0-next-20240725-cos-mt9+

![IMG_5979](https://github.com/user-attachments/assets/4274d04c-386a-461d-9e56-32be4e085049)

- wifi : works
- builtin display : works
- energy: battery status : works
- usb c and a for basic usb functionality : works
- gpu : works opengl
- bluetooth : [partial](https://github.com/hexdump0815/imagebuilder/issues/228#issuecomment-2212027148) (headphones work, keyboard broken)
- energy: dpms, suspend/resume : broken

screen doesn't full turn off, full suspens will lead to device being stuck suspended

in theory linking command
```
echo 0 | tee /sys/devices/platform/backlight-lcd0/backlight/backlight-lcd0/brightness
```
to suspens would solve screen not turning off issue
still to be figured out
- external display : broken
- hw video encode/decode : unknown
- audio : broken

_Note. tested on debian trixie_

# additional notes

again used [additional homestar config file](https://github.com/hexdump0815/imagebuilder/discussions/243#discussioncomment-10653865) when generating .config

based on latest commit (a3b832a056c9aa4b3958b0bf836461820cec9e85) from ```for-kernelci``` branch from https://gitlab.collabora.com/google/chromeos-kernel/ , 6.11 isn't there yet

additionally [wayland](https://github.com/hexdump0815/imagebuilder/issues/228#issuecomment-2210539864) still doesn't really work

this system is still very much experimental, can't likely take it a lot further on my own due to lack of experience

**Comment by hexdump0815** [comment]
@thenameisluk - thanks a lot for all the testing - i would say so far it looks not that bad, most things work and nothing seems to be completely broken so far, which is good

some things i have in mind:
- from what i remember of having read on irc or from patches i have seen it looks like external display on kukui is not ready yet and it might take a bit longer until that gets ready (lets hope for v6.12 but it might take even longer), i think they even disabled the half working display on mt8186 again for now as it made trouble until all this dp alt mode over type-c is fully sorted out ... but this topic is being worked on slowly, so at some point it will hopefully start working
- for corsola/mt8186 my plan is to try to see if it is possible to get something working with mainline v6.11 plus selectively trying to pick patches from https://patchwork.kernel.org/project/linux-mediatek/list/?series=&submitter=&state=*&q=mt8186&archive=both&delegate= ... not sure how far it is possible to get this way, but maybe there are enough patches floating around already to get it working to a certain degree ... the not booting system you saw might be just a non working display for some reason - for this a suzyqable is very helpful, as it also provides serial console output (if properly configured on the kernel cmdline) on the second of the three provided usb serial ports
- making sound work better is something which will need some work on most systems, i.e. reworking the ucm files and resolving things like the sound on your homestar like you mentioned in the other discussion ... this will take a bit of time and requires quite a bit care to not burn speakers accidently, but i hope that over the next months some progress is possible in this area too

**Reply by thenameisluk** [reply]
@hexdump0815 yep i was reading some notes with magneton having display issues, i don't really know how to configure the cmdline to make it work but will try, 'm not really sure how useful it would be

can try having look at this patches but i haven't done that before so it's kind of dark magic for me, will see

also on the speakers i know someone got them working and i doubt the homestar ones will get burned but magneton may

will try to improve as much stuff as i can and wait for **[the guide](https://github.com/hexdump0815/imagebuilder/discussions/243)** to be released ;3 would be sth useful for me to learn

**Comment by hexdump0815** [comment]
@thenameisluk - serial console via suzyqable should be something like "console=tty1 console=ttyS0,115200" on the kernel cmdline - not sure right now if it was ttyS0 or ttyS1 on the mediatek devices (experiment should bring up the correct one) - the order matters here, the last one is always the primary console (tty1 here is the normal display console btw.)

for corsola patches i'll have a look when i have some more time - in the end i would then also just try what sounds useful according to the descriptions - some patches might already be in the kernel, sometimes patches have to be adjusted slightly to patch cleanly against newer kernels than what they were made for, some patches do not work anymore at all after some time ... also following https://libera.irclog.whitequark.org/linux-mediatek/ (very low traffic, but sometimes useful information) is a good idea (https://oftc.irclog.whitequark.org/linux-cros-arm/ maybe too)

for the sound stuff it might be good to better understand how all this ucm stuff works - i also just have a rough overview and idea about it and i'm not sure if there is any really good information ... maybe reading other ucm files from https://github.com/alsa-project/alsa-ucm-conf and other similar repos with ucm files can give some ideas, also the chromeos ones might be good for inspiration, but they also might be different

regarding the kernel testing and debugging guide i at least have some skeleton already, which i'll have to fill with content now step by step

all in all things take a lot of time to get forward - it took me years to get the images to where they are right now and a lot of that was trying things, trying to understand things, git bisecting kernel regressions and finding the right patches, information or people to ask and most of the real work (patches, code etc.) is from other people and i just tried to put it all together to make it easier consumable for others in the form of bootable and useable images

**Reply by thenameisluk** [reply]
i've found magneton ucm
https://chromium.googlesource.com/chromiumos/overlays/board-overlays/+/refs/heads/main/overlay-corsola/chromeos-base/chromeos-bsp-corsola/files/magneton/audio/ucm-config/mt8186_rt1019_rt5682s.1mic.magneton/
there is a chance i won't burn my speakers

**Comment by hexdump0815** [comment]
just a little update around the v6.11 kernels: i'm right now working on bringing some more kernel trees to v6.11 (i.e. closer to the upcoming v6.12 to be better prepared for that) - i'll push what i have (i.e. marked as "coming soon" below) during the next days - these are the kernels i have at least patching and compiling well (testing still to be done):

- mt7 - chromebook_oak - is in already - https://github.com/hexdump0815/linux-mainline-mediatek-mt81xx-kernel
- mt8 - chromebook_kukui - is in already - https://github.com/hexdump0815/linux-mainline-mediatek-mt81xx-kernel
- av7 - allwinner_h3, orbsmart_s92_beelink_r89 - coming soon - https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel
- av8 - amlogic_gx - coming soon - https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel
- cbe - chromebook_snow - coming soon - https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel
- cbp - chromebook_peach - coming soon - https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel
- cbr - chromebook_veyron - coming soon - https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel

the next wave which is expected to be easy hopefully will come around a little later:

- cbg - chromebook_gru
- cbq - chromebook_trogdor (i think it was already tested sucessfully with v6.11)
- cbt - chromebook_nyan
- exy - odroid_u3
- rkc - rockchip_rk33xx
- odk - chromebook_x86_mbr, chromebook_x86_uefi, atom_x86_with_32bit_uefi

anorther wave most probably after v6.12 is out as there will be quite a bit work required around the patches, so it makes sense to wait for v6.12 first before looking into this:

- ah6 - allwinner_h6
- 616 - allwinner_h616
- r56 - rockchip_rk356x
- rpi - raspberry_pi_4
- m8x - amlogic_m8

and some kernels which will need quite a bit of work - even later:

- corsola, asurada and cherry chromebook kernels
- qcom kernels like qc7, qcn, 410, a72
- rk3588 mainline kernel
- risc-v kernels

i'll update here whenever i have merged any of those changes, i.e. when those kernels would be ready for some v6.11 testing
